cirq-core/cirq/contrib/mps_synthesis/mps_sequential_test.py

```diff
@@ -0,0 +1,333 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import numpy as np
+import pytest
+
+import cirq
+from cirq.contrib.mps_synthesis import Sequential
+
+
+def test_compile_single_qubit_state() -> None:
+    state = np.random.rand(2) + 1j * np.random.rand(2)
+    state /= np.linalg.norm(state)
+
+    encoder = Sequential()
+    circuit: cirq.Circuit = encoder(state, max_num_layers=1)
+
+    fidelity = np.vdot(cirq.final_state_vector(circuit), state)
+
+    assert np.round(np.abs(fidelity), decimals=6) == 1.0
```

Comment:
As this should be accurate, can we use `np.testing.assert_allclose(state, cirq.final_state_vector(circuit))` instead?